seatunnel-connectors/seatunnel-connector-spark-clickhouse/src/main/scala/org/apache/seatunnel/spark/sink/Clickhouse.scala

```diff
@@ -201,6 +202,7 @@ class Clickhouse extends SparkBatchSink {
     }
   }
 
+  @tailrec
```

Comment:
I think the `tailrec` annotation is only for debugging purposes, you can signal to the compiler and other developers that the method must be optimized. Can you optimize from the code level.

See [scala.annotation.tailrec](https://www.scala-lang.org/api/current/scala/annotation/tailrec.html)

> A method annotation which verifies that the method will be compiled with tail call optimization.
> If it is present, the compiler will issue an error if the method cannot be optimized into a loop.

Reply:
OK, thank you for your reply, I will try it at the code level.